crates/re_log_types/src/vec_deque_ext.rs

```diff
@@ -128,6 +208,27 @@ impl<T: Clone> VecDequeRemovalExt<T> for VecDeque<T> {
         }
         self.split_off(at)
     }
+
+    #[inline]
+    fn truncate_at(&mut self, at: usize) {
+        self.truncate(self.len() - at);
+    }
+
+    #[inline]
+    fn remove_range(&mut self, range: Range<usize>) {
+        if range.start == 0 && range.end == self.len() {
+            self.clear();
+        } else if range.start == 0 {
+            self.rotate_left(range.len());
+            self.truncate(self.len() - range.len());
+        } else if range.end == self.len() {
+            self.truncate(self.len() - range.len());
+        } else {
+            let right = self.split_off(range.end);
+            let _mid = self.split_off(range.start);
+            self.extend(right);
```

Comment:
That looks like something that could be done with a split and a truncate, saving the alloc from the second split